lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -365,12 +365,18 @@ def _warning_check(self, message_text='', expect_warning=True):
         with warnings.catch_warnings(record=True) as w:
             warnings.simplefilter('always')
             yield
-        matches = (message_text in str(warning.message) for warning in w)
-        warning_raised = any(matches)
-        msg = "Warning containing text '{}' not raised." if expect_warning \
-            else "Warning containing text '{}' unexpectedly raised."
-        self.assertEqual(expect_warning, warning_raised,
-                         msg.format(message_text))
+        warning_texts = [str(warning.message) for warning in w]
+        if expect_warning:
+            matches = [message_text in warning_text
+                       for warning_text in warning_texts]
+            warning_raised = any(matches)
+            msg = "Warning containing text '{}' not raised."
+            self.assertEqual(expect_warning, warning_raised,
+                             msg.format(message_text))
+        elif warning_texts:
+            msg = "Unexpected warnings raised : {!r}."
+            msg = msg.format(warning_texts)
+            self.assertEqual(warning_texts, [], msg)
```

Comment:
You've changed the way this works (perhaps intentionally); previously in the `expect_warning=False` case, it would only fail if a warning matching the text was raised. The reason I wrote it that way was so that if the code happened to raise some other warning it wouldn't cause the test to pass incorrectly.

Reply:
> You've changed the way this works (perhaps intentionally)

No, it's a mistake really.
I'm going to investigate (a) putting this as an assert routine in iris.tests.IrisTest, and (b) replacing the 'contains' with a Regexp